model/trace/gen-ai.yaml

```diff
@@ -0,0 +1,69 @@
+groups:
+  - id: gen_ai.request
+    type: span
+    brief: >
+      A request to an LLM is modeled as a span in a trace. The span name should be a low cardinality value representing the request made to an LLM, like the name of the API endpoint being called.
+    attributes:
+      - ref: gen_ai.system
+        requirement_level: required
+        note: >
+          If not using a vendor-supplied model, provide a custom friendly name, such as a name of the company or project.
+          If the instrumetnation reports any attributes specific to a custom model, the value provided in the `gen_ai.system` SHOULD match the custom attribute namespace segment.
+          For example, if `gen_ai.system` is set to `the_best_llm`, custom attributes should be added in the `gen_ai.the_best_llm.*` namespace.
+          If none of above options apply, the instrumentation should set `_OTHER`.
+      - ref: gen_ai.request.model
+        requirement_level: required
+        note: >
+            The name of the LLM a request is being made to. If the LLM is supplied by a vendor,
+            then the value must be the exact name of the model requested. If the LLM is a fine-tuned
+            custom model, the value should have a more specific name than the base model that's been fine-tuned.
+      - ref: gen_ai.request.max_tokens
+        requirement_level: recommended
+      - ref: gen_ai.request.temperature
+        requirement_level: recommended
+      - ref: gen_ai.request.top_p
+        requirement_level: recommended
+      - ref: gen_ai.response.id
+        requirement_level: recommended
+      - ref: gen_ai.response.model
+        requirement_level:
+          conditionally_required: if response was received
```

Comment:
In case of Azure OpenAI, I see that request model is a specific one, but response model is a generic `gpt-4`. It's also not exposed in public properties and is hard to access.

the latter makes me think that it could be of a less priority than request model and I wonder if we can downgrade the requirement level to `recommended: if available`.